botocore/handlers.py

```diff
@@ -273,6 +274,8 @@ def _needs_s3_sse_customization(params, sse_member_prefix):
 def register_retries_for_service(service_data, session,
                                  service_name, **kwargs):
     loader = session.get_component('data_loader')
+    service_id = service_data.get('metadata', {}).get('serviceId')
```

Comment:
Wonder if there's anything we can do to make this more inline with how all of the other events are setup.